File: app/src/main/java/com/osrsd/command/PrintNpcs.java
package com.osrsd.command;

import lombok.AllArgsConstructor;

import java.util.concurrent.CountDownLatch;

@AllArgsConstructor
public class PrintNpcs implements Runnable {

    private final CountDownLatch latch;

    @Override
    public void run() {
        long start = System.currentTimeMillis();
//        Library.cachePrintable(NpcDefinition.class, new NpcLoader().load().getDefinitions());
//        App.prompt(PrintNpcs.class, start);
        latch.countDown();
    }

}

File: app/src/main/java/com/osrsd/command/PrintNpcs.kt
package com.osrsd.command

import com.osrsd.cache.Library
import com.osrsd.cache.loader.ItemLoader
import com.osrsd.cache.loader.NpcLoader
import com.osrsd.spring.App
import com.osrsd.spring.domain.ItemDefinition
import com.osrsd.spring.domain.NpcDefinition
import java.util.concurrent.CountDownLatch

class PrintNpcs(val latch: CountDownLatch) : Runnable {

    override fun run() {
        val start = System.currentTimeMillis()
        Library.store(NpcDefinition::class.java, NpcLoader().load().definitions)
        App.prompt(PrintNpcs::class.java, start)
        latch.countDown()
    }

}